core/keeper.go

```diff
@@ -113,13 +115,13 @@ func triggerKeeper(evm EVMCaller) (*big.Int, error) {
 	}
 }
 
-func mint(evm EVMCaller, mintRequest *big.Int) error {
+func mint(evmCaller EVMCaller, mintRequest *big.Int, evm *vm.EVM) error {
```

Comment:
Having both `EVMCaller` and `*vm.EVM` makes no sense here. They are both the same underlying value. Instead, you should expose the related function of the `*vm.EVM` type by adding it to the `EVMCaller` interface. Also, the `GetBlockNumber()` is no longer required.